desc/objectives/_omnigenity.py

```diff
@@ -501,6 +502,232 @@ def compute(self, params, constants=None):
         return data["f_T"]
 
 
+class Omnigenity(_Objective):
+    """Omnigenity error.
+
+    This objective assumes that the collocation point (θ=0,ζ=0) lies on the contour of
+    maximum field strength |B|=B_max.
+
+    Parameters
+    ----------
+    eq : Equilibrium, optional
+        Equilibrium to be optimized to satisfy the Objective.
+    field : OmnigenousField
+        Omnigenous magnetic field to be optimized to satisfy the Objective.
+    target : float, ndarray, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        len(target) must be equal to Objective.dim_f
+    bounds : tuple, optional
+        Lower and upper bounds on the objective. Overrides target.
+        len(bounds[0]) and len(bounds[1]) must be equal to Objective.dim_f
+    weight : float, ndarray, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        len(weight) must be equal to Objective.dim_f
+    normalize : bool
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool
+       Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    grid_eq : Grid, ndarray, optional
+        Collocation grid containing the nodes to evaluate at for equilibrium data.
+    grid_field : Grid, ndarray, optional
+        Collocation grid containing the nodes to evaluate at for omnigenous field data.
+    rho : float, optional
+        Flux surface coordinate to use for the default grids.
+        Only used if `grid_eq` or `grid_field` is not supplied.
+    helicity : tuple, optional
+        Type of omnigenity (M, N). Default = quasi-isodynamic (0, 1).
+    M_booz : int, optional
+        Poloidal resolution of Boozer transformation. Default = 2 * eq.M.
+    N_booz : int, optional
+        Toroidal resolution of Boozer transformation. Default = 2 * eq.N.
+    well_weight : float, optional
+        Weight applied to the bottom of the magnetic well (B_min) relative to the top
+        of the magnetic well (B_max). Default = 1, which weights all points equally.
+    name : str
+        Name of the objective function.
+
+    """
+
+    _scalar = False
+    _linear = False
+    _units = "(T)"
+    _print_value_fmt = "Omnigenity error: {:10.3e} "
+
+    def __init__(
+        self,
+        eq,
+        field,
+        target=0,
```

Comment:
for other objectives i think we default both target and bounds to None then set a default inside init, so that either can be specified without having to override the default of the other